packages/audio/pulseaudio/package.mk

```diff
@@ -44,7 +44,7 @@ else
 fi
 
 # PulseAudio fails to build on aarch64 when NEON is enabled, so don't enable NEON for aarch64 until upstream supports it
-if echo "$TARGET_FPU" | grep -q '^neon'; then
+if [ "$TARGET_ARCH" = "arm" ] && target_has_feature neon; then
```

Comment:
Maybe "target_has_feature 32bit"? Or having an additional "target_has_features neon 32bit" looping through args and calling "target_has_feature". But that may be Overkill for just this one case.

Reply:
I'd say that adding support for combined features is maybe overkill for now.

```
if target_has_feature feat1 && target_has_feature feat2; then
```

will work in those rare cases where we need it. In addition, and would mean using a similar "pattern" when dealing with optional features:
```
if target_has_feature feat1 || target_has_feature feat2; then
```
as we couldn't handle `or` so easily in a loop (unless we seriously overcomplicate the function).

So yeah, keeping it simple may be best, and leave the combinatorial logic to the particular instance where the `target_has_feature` test is used.

I'm not sure if testing for 32bit is a good replacement for a specific arch (ie. arm) - in theory it will work as of now, but who knows if we introduce some other non-arm 32bit arch that then breaks this test? Unlikely I know, but you see my point.

Reply:
Thanks. Makes perfect sense.